### PR TITLE
⬆ Bump eslint, typescript and jest

### PR DESCRIPTION
**Pull request by dawidreedsy**
In order to build new mongoose 8 had to bump few packages and the build worked after few changes
1. 
```typescript
// src/resource.ts
if (instance === 'ObjectID' || instance === 'ObjectId') {
```
To check for the `ObjectId`, as i believe mongoose changed this. 
2. The validation error returns only the deepest property, but i do not think this is a problem as it still returns validation error. (Check `test/resource/create.spec.ts` for changes)
3. Had to update `find` function in `src/resource.ts` as sort object like {undefined: undefined} is not being removed.
4. Added `--bail` option to the `npm test` script as `jest` returns exit code `0` otherwise

### CODE REVIEW

File: package.json
Comment (alecgibson) [inline package.json:1]: You haven't bumped `typescript` as your PR title says

Comment (dawidreedsy) [reply package.json:1]: You are right didn't even notice it, this should fix the broken build

Comment (dawidreedsy) [reply package.json:1]: Changing title and description

**Comment by dawidreedsy** [comment]
@alecgibson 
I just realised Ci doesn't fail on failed tests, fixing

**Comment by dawidreedsy** [comment]
@alecgibson Seems to be ok now,
Here is the failed CiCd. 
- https://github.com/reedsy/adminjs-mongoose/actions/runs/6796432597/job/18476498543?pr=11

File: package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
   "scripts": {
     "dev": "rm -rf lib && tsc --watch",
     "build": "rm -rf lib && tsc",
-    "test": "jest --config ./test/jest.json --runInBand",
-    "cover": "jest --config ./test/jest.json --runInBand --coverage",
```
Comment (alecgibson) [inline package.json:11]: Did you just add `--bail` for testing CI failure? Can we remove please?

File: tsconfig.json [hunk 2]
```diff
@@ -10,6 +10,9 @@
     "skipLibCheck": true,
     "allowJs": true
   },
+  "include": [
+    "src/*"
```
Comment (alecgibson) [inline tsconfig.json]: Do we need this? Won't this miss build failures in `test/` files?

File: package.json (hunk 1 above)
Comment (dawidreedsy) [reply package.json:11]: Otherwise jest failes with exit code 0

Comment (dawidreedsy) [reply package.json:11]: https://github.com/reedsy/adminjs-mongoose/actions/runs/6796353512/job/18476243807

**Review by alecgibson — APPROVED** [review]

**Pull request merged**